src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -27,21 +26,38 @@ object AgoraBusiness {
     entityWithId.copy(url = Option(agoraUrl(entityWithId)))
   }
 
+  def filterByReadPermissions(authEntities: Seq[AuthorizedAgoraEntity]): Seq[AgoraEntity] = {
+    authEntities.flatMap(authEntity => filterByReadPermissions(authEntity))
+  }
+
+  def filterByReadPermissions(authEntity: AuthorizedAgoraEntity): Option[AgoraEntity] = {
+    if (authEntity.authorization.canRead) Some(authEntity.entity) else None
+  }
+
   def find(agoraSearch: AgoraEntity,
            agoraProjection: Option[AgoraEntityProjection],
            entityTypes: Seq[AgoraEntityType.EntityType]): Seq[AgoraEntity] = {
-    AgoraDao.createAgoraDao(entityTypes).find(agoraSearch, agoraProjection).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    val entities = AgoraDao.createAgoraDao(entityTypes).find(agoraSearch, agoraProjection).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities))
   }
 
   def findSingle(namespace: String,
                  name: String,
                  snapshotId: Int,
                  entityTypes: Seq[AgoraEntityType.EntityType]): Option[AgoraEntity] = {
-    AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId).map(entity => entity.copy(url = Option(agoraUrl(entity))))
+    val optEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId).map(entity => entity.copy(url = Option(agoraUrl(entity))))
```

Comment:
We could though it clutters things a little since you would be doing the match here as well as below.